docs/Morphir CLI incremental build approach (ADR format).md

```diff
@@ -0,0 +1,95 @@
+# ADR for Morphir CLI incremental build approach
```

Comment:
Can you change the name of the file to follow the naming convention like this:
`0002-morphir-cli-incremental-build-approach.md`